Sparkle/InstallerProgress/InstallerProgressAppController.m

```diff
@@ -193,8 +193,21 @@ - (void)cleanupAndExitWithStatus:(int)status error:(NSError * _Nullable)error __
     
     // Remove the agent bundle; it is assumed this bundle is in a temporary/cache/support directory
     NSError *theError = nil;
-    if (![[NSFileManager defaultManager] removeItemAtPath:[[NSBundle mainBundle] bundlePath] error:&theError]) {
+    NSString *bundlePath = [[NSBundle mainBundle] bundlePath];
+    
+    if (![[NSFileManager defaultManager] removeItemAtPath:bundlePath error:&theError]) {
         SULog(SULogLevelError, @"Couldn't remove agent bundle: %@.", theError);
+    } else {
+        // There should be nothing else in the parent temporary directory given to us,
+        // so let us try to remove it. Note rmdir() will fail if there are unexpectably other
+        // items present
+        NSString *parentDirectory = bundlePath.stringByDeletingLastPathComponent;
+        const char *fileSystemRepresentation = parentDirectory.fileSystemRepresentation;
+        if (fileSystemRepresentation != NULL) {
+            if (rmdir(fileSystemRepresentation) != 0) {
```

Comment:
why isn't FileManager handling this properly?

Reply:
Framework originally creates this temp directory and the agent app bundle inside it, but the progress agent / relaunched tool destroys it on cleanup. Since it is "transferred", the agent only removes the temp directory (after removing the agent app in it) if it's empty, which rmdir() does. Just a safety precaution. I could add a non-recursive dir removal method to SUFileManager but I don't consider that too important.